packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -1156,6 +1161,116 @@ class EndpointIndex<TInterval extends ISerializableInterval> implements Interval
 	}
 }
 
+export class EndInRangeIndex<TInterval extends ISerializableInterval>
+	implements IntervalIndex<TInterval>
+{
+	private readonly intervalTree;
+
+	constructor(
+		private readonly helpers: IIntervalHelpers<TInterval>,
+		private readonly client?: Client,
```

Comment:
I'd prefer to type `client` as being required. you'll need to move where you cast around, but it should make the cast more colocated with where the non-sequence helpers are declared (which makes it more obvious the cast is acceptable)